Git Beta: Git Pull failing show surface errors from git

### Summary

Git pull failure should surface the error message in Zed.

### Description

When a pull fails like so:
```
~/zcode/zed (main) $ git pull
Updating 9b07f36199..fcc5e27455
error: Your local changes to the following files would be overwritten by merge:
	assets/keymaps/default-linux.json
Please commit your changes or stash them before you merge.
Aborting
```

Actual Behavior: Zed does not surface the error in toast, nor in logs.

Expected Behavior: Zed show the failure output from `git pull`.

<details><summary>Screen recording of remote</summary>

https://github.com/user-attachments/assets/d53cd92e-08b0-4d5b-a031-183dd2643497

</details>

Steps to reproduce are a little obtuse, but here we go:

```
cd /tmp/
git clone git@github.com:zed-industries/zed zed2
cd zed2
git checkout -b foobar
git push 
cd ..

git clone git@github.com:zed-industries/zed zed3
cd zed3
git checkout foobar
echo foo >> README.md
git add README.md
git commit -m "foo"
git push
cd ..

cd zed2
echo bar >> README.md
git pull
# see cli error message
zed .
# then click pull in Zed
```

### Zed Version and System Specs

Zed: v0.178.0 (Zed Nightly a887f3b340b8844d4b489ac785a75e7e939389c3) 
OS: macOS 15.3.1
Memory: 64 GiB
Architecture: aarch64

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.

Locally there is no log.
Remote has this:
```
2025-03-07T10:40:09.872979-05:00 [ERROR] (remote server) server:error handling message. type:Pull, error:Operation failed:
```

--- a/crates/git/src/repository.rs
+++ b/crates/git/src/repository.rs
@@ -692,7 +692,9 @@ impl GitRepository for RealGitRepository {
                 PushOptions::Force => "--force-with-lease",
             }))
             .arg(remote_name)
-            .arg(format!("{}:{}", branch_name, branch_name));
+            .arg(format!("{}:{}", branch_name, branch_name))
+            .stdout(smol::process::Stdio::piped())
+            .stderr(smol::process::Stdio::piped());
         let git_process = command.spawn()?;
 
         run_remote_command(ask_pass, git_process)
@@ -714,7 +716,9 @@ impl GitRepository for RealGitRepository {
             .current_dir(&working_directory)
             .args(["pull"])
             .arg(remote_name)
-            .arg(branch_name);
+            .arg(branch_name)
+            .stdout(smol::process::Stdio::piped())
+            .stderr(smol::process::Stdio::piped());
         let git_process = command.spawn()?;
 
         run_remote_command(ask_pass, git_process)
@@ -729,7 +733,9 @@ impl GitRepository for RealGitRepository {
             .env("SSH_ASKPASS", ask_pass.script_path())
             .env("SSH_ASKPASS_REQUIRE", "force")
             .current_dir(&working_directory)
-            .args(["fetch", "--all"]);
+            .args(["fetch", "--all"])
+            .stdout(smol::process::Stdio::piped())
+            .stderr(smol::process::Stdio::piped());
         let git_process = command.spawn()?;
 
         run_remote_command(ask_pass, git_process)

--- a/crates/git_ui/src/git_panel.rs
+++ b/crates/git_ui/src/git_panel.rs
@@ -1571,6 +1571,7 @@ impl GitPanel {
                         this.show_remote_output(RemoteAction::Fetch, remote_message, cx);
                     }
                     Err(e) => {
+                        log::error!("Error while fetching {:?}", e);
                         this.show_err_toast(e, cx);
                     }
                 }
@@ -1629,7 +1630,10 @@ impl GitPanel {
                 Ok(remote_message) => {
                     this.show_remote_output(RemoteAction::Pull, remote_message, cx)
                 }
-                Err(err) => this.show_err_toast(err, cx),
+                Err(err) => {
+                    log::error!("Error while pull {:?}", err);
+                    this.show_err_toast(err, cx)
+                }
             })
             .ok();
 
@@ -1697,6 +1701,7 @@ impl GitPanel {
                     this.show_remote_output(RemoteAction::Push(remote), remote_message, cx);
                 }
                 Err(e) => {
+                    log::error!("Error while pushing {:?}", e);
                     this.show_err_toast(e, cx);
                 }
             })?;

